physical/dynamodb.go

```diff
@@ -378,6 +378,36 @@ func (d *DynamoDBBackend) List(prefix string) ([]string, error) {
 	return keys, nil
 }
 
+func (d *DynamoDBBackend) hasChildren(prefix string) (bool, error) {
+	defer metrics.MeasureSince([]string{"dynamodb", "list"}, time.Now())
+
+	prefix = strings.TrimSuffix(prefix, "/")
+
+	prefix = escapeEmptyPath(prefix)
+	queryInput := &dynamodb.QueryInput{
+		TableName:      aws.String(d.table),
+		ConsistentRead: aws.Bool(true),
+		KeyConditions: map[string]*dynamodb.Condition{
+			"Path": {
+				ComparisonOperator: aws.String("EQ"),
+				AttributeValueList: []*dynamodb.AttributeValue{{
+					S: aws.String(prefix),
+				}},
+			},
+		},
+		Limit: aws.Int64(2),
```

Comment:
I think you should document why the limit is 2 and not 1 (in a comment I'd guess)